server.js

```diff
@@ -1,26 +1,78 @@
 var http = require('http');
 var https = require('https');
+var fs = require('fs');
 var path = require('path');
 var express = require('express');
+var helmet = require('helmet');
+var bodyParser = require('body-parser');
+var crypto = require('crypto');
 
 var config = require('./config.server.js');
 
 var buildDir = 'dist';
 
 var app = express();
-var helmet = require('helmet');
+
+var nonceMiddleware = function (req, res, next) {
+  // Cache static html file to avoid reading it from the filesystem on each request
+  if (!global.html) {
+    console.log('Caching static HTML');
+    global.html = fs.readFileSync(staticDir + '/index.html', 'utf8');
```

Comment:
Recommend template literals for readability:
`fs.readFileSync(staticDir + '/index.html', 'utf8');` -> `` fs.readFileSync(`${staticDir}/index.html`, 'utf8'); ``